### ISSUE TITLE
Compatibility with Pro-builder pb_objects

### ISSUE THREAD

**Issue by jeffvella** [Whinarn/UnityMeshSimplifier#9]
I am struggling to get this work correctly with pro-builder objects; with pro-builder being integrated into Unity core it will be important going forward. Any chance you could look into it?

**Comment by Whinarn** [comment]
I have not performed any tests with pro-builder meshes yet.
Do the results look bad or are you having problems getting it to work at all?

I will try to make some time to test it this weekend or the coming week.
Let me get back to you once I have tested it.

**Comment by Whinarn** [comment]
Hello again, @imxzjv !
I tested UnityMeshSimplifier with Unity 2018.2.0f2 and ProBuilder, and did experience some UV issues and two other non-related minor bugs that was easy solvable.

I tried to dig into why the UV issue occurred, but it's getting late so I will have to resume another day.
Could you possibly give me any further details into what you are not getting to work correctly? Screenshots or samples?

In short, the UV issue resulted in two faces that had merged turned into very scewed UVs. But only on a few places where I tested it. Is this the same problem as you were having?

Or are you talking about the fact that ProBuilder sometimes regenerates the mesh
 and gets rid of the decimated result?

Thanks!

**Comment by imxzjv** [comment]
Heya, thanks for taking the time to look into it. I will try to put some examples together. 

I took the CSG  code out of the latest pro-builder 'experimental boolean tool' with reflector and have been using it to cut holes in pro builder objects. 

The problem for me is that CSG adds a bunch of unnecessary complexity to the model, and after a few operations it starts to get ridiculous (growing from 5k tris to 50k+). So I tried to use your tool to clean up the mess (preferably without effecting the geometry). What i was seeing is an imploding/exploding of the model after running mesh simplifier on it. 

But i now realize that its probably because the pb_object created from CSG is not just messy, but also mal-formed. If i try to use the grow-selection tool on a face, it can't select everything which means the 'winged-edges' scan is failing with a bunch of edges not being linked properly with their opposite face. 

So unless this is something you're interested in, probably a bit of an edge case. :)

**Comment by Whinarn** [comment]
Since I don't have a lot of spare time over for a week or two now, I would greatly appreciate if you could provide a basic project with one of these meshes that you would experience a problem with, and I can have a closer look of what is going on with the algorithm.

**Comment by Whinarn** [comment]
Since there has not been any activity here and the fact that I could not reproduce the problem mentioned, I will close this issue. Please re-open if anyone else has the same problem. And I would greatly appreciate if examples could be provided.

The UV problem that I mentioned has been improved with the use of barycentric attribute interpolation.
